c_src/xav/reader.c

```diff
@@ -175,6 +184,24 @@ int reader_next_frame(struct Reader *reader) {
   return 0;
 }
 
+int reader_seek_to_time(struct Reader *reader, double time_in_seconds) {
+    int64_t frmseekPos = av_rescale_q(
+        (int64_t)(time_in_seconds * AV_TIME_BASE),
+        AV_TIME_BASE_Q,
+        reader->fmt_ctx->streams[reader->stream_idx]->time_base
+    );
```

Comment:
I think that what we need to do here is to convert time_in_seconds to frame number. Given that time_base is 1/fps we should do time_in_seconds * time_base.den / time_base.num so :
```suggestion
av_rescale(time_in_seconds, time_base.den, time_base.num);
```

Reply:
not exactly..`av_seek_frame` expects the seeking position in the same timebase as the selected stream not the frame number. I did some further research and it seems, that this is the correct way to calculate the seeking position and preserving the float accuracy. For example the file in `test/fixtures/sample_h264.mp4` has a timebase of 1/15360 and `AV_TIME_BASE` is 1000000

Reply:
Ohh okay :man_facepalming:  Could you please add a comment or link some resource that explains what we do here? I mean, what's the difference between AV_TIME_BASE and AVStream.time_base and why we need to use both in this calculation?

Reply:
Absolutely!